src/App.jsx

```diff
@@ -14,45 +15,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+
+function getPreparedGoods(
+  goods,
+  { sortBy = SORT_BY_ALPHABET,
+    isSortRevers = false },
+) {
+  let preparedGoods = [...goods];
+
+  switch (sortBy) {
+    case SORT_BY_ALPHABET:
+      preparedGoods = preparedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SORT_BY_LENGTH:
+      preparedGoods = preparedGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      break;
+  }
+
+  return isSortRevers ? preparedGoods.reverse() : preparedGoods;
+}
+
+export const App = () => {
+  const [sortBy, setSortBy] = useState('');
+  const [isSortRevers, setIsSortRevers] = useState(false);
+  const handlerSortByAlphabet = () => {
+    setSortBy(sortBy === SORT_BY_ALPHABET
+      ? ''
+      : SORT_BY_ALPHABET);
+  };
+
+  const handlerSortByLength = () => {
+    setSortBy(sortBy === SORT_BY_LENGTH
+      ? ''
+      : SORT_BY_LENGTH);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handlerSortRevers = () => {
+    setIsSortRevers(isSortRevers === false);
+  };
+
+  const handlerSortReset = () => {
```

Comment:
function names should start with a verb
```suggestion
  const handleSortReset = () => {
```